### ISSUE TITLE
False positive - superfluous parens

### ISSUE THREAD

**Issue by reconndev** [PyCQA/pylint#5779]
### Bug description

The parentheses are essential for this to work:
```python3
from pathlib import Path
paths = [Path('-')]
if (stdin := Path('-')) in paths:
  assert isinstance(stdin, pathlib.Path)
```
Otherwise:
```python3
from pathlib import Path
paths = [Path('-')]
if stdin := Path('-') in paths:
  assert isinstance(stdin, pathlib.Path)  # Assertion fails here
```

### Configuration

```ini
Default
```

### Command used

```shell
python3 -m pylint file1.py
```

### Pylint output

```shell
************* Module mymodule.file1
file1.py:23:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)

------------------------------------------------------------------
Your code has been rated at 9.87/10 (previous run: 9.87/10, +0.00)
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 2.6.0
astroid 2.4.2
Python 3.9.10 (main, Jan 15 2022, 11:40:53)
[Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

Mac OS (M1 processor)

### Additional dependencies

N/A

**Comment by DanielNoord** [comment]
This no longer raises on `main`. I think it is also fixed in `2.12`. Are you able to update your version of `pylint`?

**Comment by reconndev** [comment]
My bad. Should have checked the latest version. Thanks for your prompt response!
